### DISCUSSION TITLE
cors not working?

### DISCUSSION THREAD

**Discussion by fockarty** [miguelgrinberg/microdot#219] [Q&A] [answered]
Hi, i am trying to make something simple but i run into cors issues when i post something to /control the browser gives an cors error, in the response headers there in no mention of allow origins headers (or any response headers). What am i missing here?

from microdot import Microdot, send_file, Request,Response
from microdot.cors import CORS

import json

app = Microdot()
cors=CORS(app, allowed_origins=['*'], allow_credentials=True,handle_cors=True)

@app.route('/')
async def index(request):
    return send_file('Sockets/web/index.html')

@app.route('/favicon.ico')
async def favicon(request):
    return send_file('Sockets/web/favicon.ico')

@app.route('/assets/<file>')
async def assets(request,file):
   return send_file('Sockets/web/assets/'+file)

@app.route('/control/',methods=["GET", "POST","OPTIONS"])
async def assets(request):
   obj=json.loads(request.body.decode(encoding='UTF-8'))
   red=obj['kleur'][0]['r']
   blue=obj['kleur'][0]['b']
   green=obj['kleur'][0]['g']

   print(f'{ red } { green } { blue } ' )
   return Response(body="ok",status_code=200)

@app.route('/shutdown')
async def shutdown(request):
    request.app.shutdown()
    return 'The server is shutting down...'

app.run(debug=True)

**Comment by fockarty** [comment]
the response headers:

HTTP/1 200 OK
Allow: GET, POST, OPTIONS, HEAD, OPTIONS
Access-Control-Allow-Methods: POST
Access-Control-Allow-Headers: content-type,test
Content-Length: 0
Content-Type: text/plain; charset=UTF-8

**Reply by miguelgrinberg** [reply]
What's the CORS error? That is very important, there are many things that can fail with CORS.

**Comment by fockarty** [comment]
and solved the example says  allowed_origins=['https://example.org'] but in case of * it shoul be  allowed_origins='*'. dunno if this is always the case

**Reply by miguelgrinberg** [reply, marked as answer]
That is correct. A list of domains or `'*'` for all domains.

**Reply by fockarty** [reply]
cool, thanks that makes thing clear :-)
